test/integration/admin/test_admin_call.py

```diff
@@ -0,0 +1,164 @@
+        '--name', project.name,
+        '--run-dir', run_dir,
+        'func_bad_arg_type', '--bad-arg', 'some-value',
+    ]
+    rc, output = run_command_and_get_output(cmd, cwd=tmpdir)
+    assert rc == 1
+
+    assert 'Admin function "func_bad_arg_type" accepts value of unsupported type: bad-type' in output
+
+
+def test_func_long_arg(cartridge_cmd, custom_admin_running_instances, tmpdir):
+    project = custom_admin_running_instances['project']
+    run_dir = custom_admin_running_instances['run_dir']
+
+    cmd = [
+        cartridge_cmd, 'admin',
+        '--name', project.name,
+        '--run-dir', run_dir,
+        'func_long_arg', '--long-arg', 'some-value',
+    ]
+    rc, output = run_command_and_get_output(cmd, cwd=tmpdir)
+    assert rc == 0
+
+    assert get_log_lines(output) == [
+        '• func_long_arg was called with "some-value" arg',
+    ]
+
+
+def test_func_rets_str(cartridge_cmd, custom_admin_running_instances, tmpdir):
+    project = custom_admin_running_instances['project']
+    run_dir = custom_admin_running_instances['run_dir']
+
+    cmd = [
+        cartridge_cmd, 'admin',
+        '--name', project.name,
+        '--run-dir', run_dir,
+        'func_rets_str',
+    ]
+    rc, output = run_command_and_get_output(cmd, cwd=tmpdir)
+    assert rc == 0
+
+    assert get_log_lines(output) == [
+        '• func_rets_str was called',
+    ]
+
+
+def test_func_rets_non_str(cartridge_cmd, custom_admin_running_instances, tmpdir):
+    project = custom_admin_running_instances['project']
+    run_dir = custom_admin_running_instances['run_dir']
+
+    cmd = [
+        cartridge_cmd, 'admin',
+        '--name', project.name,
+        '--run-dir', run_dir,
+        'func_rets_non_str',
+    ]
+    rc, output = run_command_and_get_output(cmd, cwd=tmpdir)
+    assert rc == 0
+
+    assert get_log_lines(output) == [
+        '• 666',
+        '• Admin function should return string or string array value',
+    ]
+
+
+def test_func_rets_err(cartridge_cmd, custom_admin_running_instances, tmpdir):
+    project = custom_admin_running_instances['project']
+    run_dir = custom_admin_running_instances['run_dir']
+
+    cmd = [
+        cartridge_cmd, 'admin',
+        '--name', project.name,
+        '--run-dir', run_dir,
+        'func_rets_err',
+    ]
+    rc, output = run_command_and_get_output(cmd, cwd=tmpdir)
+    assert rc == 1
+
+    assert get_log_lines(output) == [
+        '⨯ Failed to call "func_rets_err": Some horrible error',
+    ]
+
+
+def test_func_raises_err(cartridge_cmd, custom_admin_running_instances, tmpdir):
+    project = custom_admin_running_instances['project']
+    run_dir = custom_admin_running_instances['run_dir']
+
+    cmd = [
+        cartridge_cmd, 'admin',
+        '--name', project.name,
+        '--run-dir', run_dir,
+        'func_raises_err',
+    ]
+    rc, output = run_command_and_get_output(cmd, cwd=tmpdir)
+    assert rc == 1
+
+    assert '⨯ Failed to call "func_raises_err":' in output
```

Comment:
I suggest to exclude `⨯` from output check.

This symbol could be easily changed.

Reply:
I have no idea why this symbol should be changed. I don't see any reason to remove it.